src/netguard/tls.c

```diff
@@ -59,7 +59,7 @@ void get_server_name(
         } else if (content_type == TLS_TYPE_HANDSHAKE_RECORD) { // content type handshake
             // handshake packet type
             uint16_t tls_handshake_size = (tls[3] << 8 & 0xFF00) + (tls[4] & 0x00FF);
-            if (length - (tls - pkt) < 5) {
+            if (length - (tls - pkt) < tls_handshake_size + 4) {
```

Comment:
I am not sure I understand the change.

The original check was to ensure the TLS header had a record header (ie. 5 bytes).
Why would that check cause a wrong index in line 78?

Reply:
I don't think the change is necessary. Regardless, it also doesn't seem correct, `tls_handshake_size + 4` is one byte short. The TLS record header has 5 bytes (1 type + 2 version + 2 size of handshake message).

Reply:
Thanks for catching my off-by-one error - fixed. Let's merge this and see if it fixes the crashes, then go from there.